src/braket/emulators/emulator_passes/criteria/gate_connectivity_criterion.py

```diff
@@ -0,0 +1,121 @@
+from typing import Any, Dict, Iterable, Tuple, Union
+
+from networkx import DiGraph
+from networkx.utils import graphs_equal
+
+from braket.circuits.circuit import Circuit
+from braket.circuits.compiler_directives import EndVerbatimBox, StartVerbatimBox
+from braket.circuits.gate import Gate
+from braket.emulators.emulator_passes.criteria.emulator_criterion import EmulatorCriterion
+from braket.registers.qubit_set import QubitSet
+
+
+class GateConnectivityCriterion(EmulatorCriterion):
+    def __init__(
```

Comment:
1. The scenario you've described should be an error case, if `directed=False` reverse edges should have the same supported gates or should not be provided (and later filled in). 

2. While no current gate-based QPUs have directed edges with asymmetric supported gate sets, supporting directed graphs was done in consideration of QPUs like OQC's Lucy, which has a directed topology - because QPUs with directed topologies _have_ been supported by Braket, I felt it necessary to support this for future cases. Additionally, I had considered a case where we might _remove_ supported gates in certain directions if, for instance, the fidelity of a gate applied in one direction was far worse than in the other direction.